src/charm.py

```diff
@@ -276,14 +276,10 @@ def _on_loki_pebble_ready(self, _):
             logger.debug("Cannot set workload version at this time: could not get Loki version.")
 
     def _on_node_exporter_pebble_ready(self, _):
-        current_layer = self._node_exporter_container.get_plan()
         new_layer = self._node_exporter_pebble_layer
-        restart = current_layer.services != new_layer.services
-
-        if restart:
-            self._node_exporter_container.add_layer("node-exporter", new_layer, combine=True)
-            self._node_exporter_container.restart("node-exporter")
-            logger.info("Node Exporter (re)started")
+        self._node_exporter_container.add_layer("node-exporter", new_layer, combine=True)
+        self._node_exporter_container.replan()
+        logger.info("Node Exporter (re)started")
```

Comment:
not sure if 'restarted' is correct here. We're adding a layer that didn't exist previously, so probably "started" is enough.